pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -1651,3 +1682,22 @@ function peerDependenciesWithoutOwn (pkg: PackageManifest): PeerDependencies {
   }
   return result
 }
+
+function getCatalogReplacementPref (
```

Comment:
Moved this code out of `resolveDependenciesOfImporterDependency` for easier reuse.